spec/models/configuration_script_spec.rb

```diff
@@ -34,10 +34,48 @@
     end
 
     it "accepts different variables to launch a job template against" do
-      added_extras = {'extra_vars' => {'some_key' => 'some_value'}.to_json}
+      added_extras = {:extra_vars => {:some_key => :some_value}}
       expect(manager.configuration_scripts.first.run(added_extras)).to be_a AnsibleTowerClient::Job
-      expect_any_instance_of(AnsibleTowerClient::JobTemplate).to receive(:launch).with(added_extras)
+      expect_any_instance_of(AnsibleTowerClient::JobTemplate).to receive(:launch).with(:extra_vars=>"{\"instance_ids\":[\"i-3434\"],\"some_key\":\"some_value\"}")
       manager.configuration_scripts.first.run(added_extras)
     end
+
+    it "merges a json string and a hash to send out to the tower gem" do
+      external = {:some_key => :some_value}
+      internal = manager.configuration_scripts.first.variables
+      config_script = manager.configuration_scripts.first
```

Comment:
@syncrou should the config_script line and internal line be swapped so you can get the variables from config_script
config_script = manager.configuration_scripts.first
internal = config_script.variables